splash/qtutils.py

```diff
@@ -141,9 +140,19 @@ def qurl2ascii(url):
     return url
 
 
+def to_qurl(s):
+    if isinstance(s, QUrl):
+        return s
+
+    if isinstance(s, unicode):
+        s = s.encode('utf-8')
+
+    return QUrl.fromEncoded(s)
```

Comment:
Examining the source here:
http://cep.xray.aps.anl.gov/software/qt4-x11-4.8.6-browser/d4/dd4/qurl_8cpp_source.html#l04443
it manipulates a char array and percent-encode any non-safe characters, so it will work if you pass a utf-8 encoded char array.